### ISSUE TITLE
Failed to handle STREAM_MUTATION_FRAGMENTS when interrupting rebuild operation with a hard reboot

### ISSUE THREAD

**Issue by KnifeyMoloko** [scylladb/scylla#10746]
## Installation details

Kernel Version: 5.13.0-1025-aws
Scylla version (or git commit hash): `5.1.dev-20220606.605ee74c39b2` with build-id `8d11831b4f5157cf34644edbc6314947acebd2b7`
Cluster size: 6 nodes (i3.4xlarge)

Scylla Nodes used in this run:
- longevity-100gb-4h-master-db-node-d24e64a0-9 (34.244.15.71 | 10.0.3.251) (shards: 14)
- longevity-100gb-4h-master-db-node-d24e64a0-8 (63.34.20.145 | 10.0.1.0) (shards: 14)
- longevity-100gb-4h-master-db-node-d24e64a0-7 (52.49.242.119 | 10.0.2.50) (shards: 14)
- longevity-100gb-4h-master-db-node-d24e64a0-6 (54.171.106.200 | 10.0.3.168) (shards: 14)
- longevity-100gb-4h-master-db-node-d24e64a0-5 (52.210.19.254 | 10.0.1.71) (shards: 14)
- longevity-100gb-4h-master-db-node-d24e64a0-4 (34.246.187.230 | 10.0.0.21) (shards: 14)
- longevity-100gb-4h-master-db-node-d24e64a0-3 (34.245.193.188 | 10.0.2.146) (shards: 14)
- longevity-100gb-4h-master-db-node-d24e64a0-2 (34.240.103.114 | 10.0.1.31) (shards: 14)
- longevity-100gb-4h-master-db-node-d24e64a0-1 (34.243.107.29 | 10.0.0.11) (shards: 14)

OS / Image: `ami-05e0cc035a45d963a` (aws: eu-west-1)

Test: `longevity-100gb-4h-test`
Test id: `d24e64a0-00d8-4d7f-943b-0f25a306209c`
Test name: `scylla-master/longevity/longevity-100gb-4h-test`
Test config file(s):

- [longevity-100gb-4h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/5c1a8968e7a2c1bbbca61fa40c6da380a40cd939/test-cases/longevity/longevity-100gb-4h.yaml)

## Issue description

**>>>>>>>**
While running `RebuildStreamingErr` nemesis we perform the following steps:
1.  Stop Scylla on the target node.
2. Delete some of the non-system data files.
3. Start a `rebuild` operation on the target node.
4. Interrupt `rebuild` with a hard reboot of the target node.

After step 4, after triggering the interrupting reboot, we're seeing errors:

```
Jun 07 02:48:01 longevity-100gb-4h-master-db-node-d24e64a0-1 scylla[29363]:  [shard 4] rpc - client 10.0.3.251:62654: unexpected eof on a stream while reading data: expected 5609 got 2048
Jun 07 02:48:01 longevity-100gb-4h-master-db-node-d24e64a0-1 scylla[29363]:  [shard 4] rpc - client 10.0.3.168:55178: unexpected eof on a stream while reading data: expected 5609 got 4728
Jun 07 02:48:01 longevity-100gb-4h-master-db-node-d24e64a0-1 scylla[29363]:  [shard 6] rpc - client 10.0.2.146:52436: unexpected eof on a stream while reading data: expected 5609 got 3178
Jun 07 02:48:01 longevity-100gb-4h-master-db-node-d24e64a0-1 scylla[29363]:  [shard 6] stream_session - [Stream #3e0c43f0-e60c-11ec-8499-16012f133edc] Failed to handle STREAM_MUTATION_FRAGMENTS (receive and distribute phase) for ks=keyspace1, cf=standard1, peer=10.0.1.71: seastar::nested_exception: seastar::rpc::stream_closed (rpc stream was closed by peer) (while cleaning up after seastar::rpc::stream_closed (rpc stream was closed by peer))
Jun 07 02:48:01 longevity-100gb-4h-master-db-node-d24e64a0-1 scylla[29363]:  [shard 6] stream_session - [Stream #3e0c43f0-e60c-11ec-8499-16012f133edc] Failed to handle STREAM_MUTATION_FRAGMENTS (respond phase) for ks=keyspace1, cf=standard1, peer=10.0.1.71: seastar::rpc::closed_error (connection is closed)
Jun 07 02:48:01 longevity-100gb-4h-master-db-node-d24e64a0-1 scylla[29363]:  [shard 10] stream_session - [Stream #3e0c43f0-e60c-11ec-8499-16012f133edc] Failed to handle STREAM_MUTATION_FRAGMENTS (receive and distribute phase) for ks=keyspace1, cf=standard1, peer=10.0.1.71: seastar::nested_exception: seastar::rpc::stream_closed (rpc stream was closed by peer) (while cleaning up after seastar::rpc::stream_closed (rpc stream was closed by peer))
Jun 07 02:48:01 longevity-100gb-4h-master-db-node-d24e64a0-1 scylla[29363]:  [shard 10] stream_session - [Stream #3e0c43f0-e60c-11ec-8499-16012f133edc] Failed to handle STREAM_MUTATION_FRAGMENTS (respond phase) for ks=keyspace1, cf=standard1, peer=10.0.1.71: seastar::rpc::closed_error (connection is closed)
Jun 07 02:48:01 longevity-100gb-4h-master-db-node-d24e64a0-1 scylla[29363]:  [shard 9] rpc - client 10.0.2.146:53965: unexpected eof on a stream while reading data: expected 5609 got 5390
```

The target node does look to recover - after the reboot it triggers a rebuild and seems to be able to finish it without issues.
**<<<<<<<**

- Restore Monitor Stack command: `$ hydra investigate show-monitor d24e64a0-00d8-4d7f-943b-0f25a306209c`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=d24e64a0-00d8-4d7f-943b-0f25a306209c)
- Show all stored logs command: `$ hydra investigate show-logs d24e64a0-00d8-4d7f-943b-0f25a306209c`

## Logs:
- **db-cluster-d24e64a0.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/d24e64a0-00d8-4d7f-943b-0f25a306209c/20220607_030413/db-cluster-d24e64a0.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/d24e64a0-00d8-4d7f-943b-0f25a306209c/20220607_030413/db-cluster-d24e64a0.tar.gz)
- **monitor-set-d24e64a0.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/d24e64a0-00d8-4d7f-943b-0f25a306209c/20220607_030413/monitor-set-d24e64a0.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/d24e64a0-00d8-4d7f-943b-0f25a306209c/20220607_030413/monitor-set-d24e64a0.tar.gz)
- **loader-set-d24e64a0.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/d24e64a0-00d8-4d7f-943b-0f25a306209c/20220607_030413/loader-set-d24e64a0.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/d24e64a0-00d8-4d7f-943b-0f25a306209c/20220607_030413/loader-set-d24e64a0.tar.gz)
- **sct-runner-d24e64a0.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/d24e64a0-00d8-4d7f-943b-0f25a306209c/20220607_030413/sct-runner-d24e64a0.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/d24e64a0-00d8-4d7f-943b-0f25a306209c/20220607_030413/sct-runner-d24e64a0.tar.gz)
- **parallel-timelines-report-d24e64a0.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/d24e64a0-00d8-4d7f-943b-0f25a306209c/20220607_030413/parallel-timelines-report-d24e64a0.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/d24e64a0-00d8-4d7f-943b-0f25a306209c/20220607_030413/parallel-timelines-report-d24e64a0.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-100gb-4h-test/460/)

**Comment by slivne** [comment]
because of the hard reboot the streams that have started are not cleared cleanly - we can see the error states that ```(while cleaning up after seastar::rpc::stream_closed (rpc stream was closed by peer)```

I think this is not pretty but fine - is this a regression ?

**Comment by slivne** [comment]
@KnifeyMoloko ^^ ?

**Comment by KnifeyMoloko** [comment]
@slivne I don't think it's a regression. We've recently made some changes to make `DecommissionStreamingErr` more aggressive with how it interrupts the decommission process and that's what made us trigger this error.

**Comment by fruch** [comment]
Issue moved to [scylladb/scylla-cluster-tests #4936](https://github.com/scylladb/scylla-cluster-tests/issues/4936) via [**ZenHub**](https://www.zenhub.com/)

**Comment by roydahan** [comment]
@KnifeyMoloko I don't understand what the issue is...
The fact that we got a new error during a stopped rebuild?
It sounds like it's expected, if so, please close the issue.

**Comment by KnifeyMoloko** [comment]
@roydahan Yes, this was a new error we saw during interrupting the rebuild process. Since we have a follow-up issue in SCT, I'm closing this one.

**Comment by fruch** [comment]
@KnifeyMoloko what is the issue  number of that follow up ? (since we want to reference it in places we encounter it)
